src/xregexp.js

```diff
@@ -1792,7 +1792,7 @@ XRegExp.addToken(
  * and $ only. Also allows numbered backreferences as `\k<n>`.
  */
 XRegExp.addToken(
-    /\\k<([\w$]+)>/,
+    /\\k<([\p{ID_Start}$_][\p{ID_Continue}$_\u200C\u200D]*)>/u,
```

Comment:
This change inappropriately drops support for numbered backreferences.

Also, given that the handler function for this token already throws a `SyntaxError` if the backreference refers to an capturing group that doesn't appear to the left within the same regex, this doesn't need to spell out the rules like this (given that `\p{ID_Start}` and `\p{ID_Continue}` get expanded into thousands of characters.

I'd recommend simplifying into `/\\k<([^>]+)>/`.

This does not change XRegExp regex syntax at all thanks to XRegExp's syntax strictness. In regexes, things starting with `\k` that don't fully match the original regex (`/\\k<([\w$]+)>/`) will already cause an error (`SyntaxError: Invalid escape \k`).

Reply:
Ooh, that's neat, about how we only need to strictly check the characters name when the capture group is parsed, and not when the backreference gets parsed. Great optimization there, I've updated it as suggested.